File: src/main/java/io/chesslave/recognition/Colors.java
package io.chesslave.recognition;

import java.awt.*;

public class Colors {

    public static boolean areSimilar(Color a, Color b) {
        return Math.abs(hue(a) - hue(b)) < .1;
    }

    public static float hue(Color color) {
        final float hsb[] = new float[3];
        Color.RGBtoHSB(color.getRed(), color.getGreen(), color.getBlue(), hsb);
        return hsb[0];
    }
}

File: backend/src/main/java/io/chesslave/eyes/Colors.kt
package io.chesslave.eyes

import java.awt.Color

object Colors {

    @JvmStatic @JvmOverloads
    fun areSimilar(a: Color, b: Color, tolerance: Double = .02): Boolean = Math.abs(hue(a) - hue(b)) < tolerance

    private fun hue(color: Color): Float {
        val hsb = FloatArray(3)
        Color.RGBtoHSB(color.red, color.green, color.blue, hsb)
        return hsb[0]
    }
}